test/command/suite/select/query/match/with_index/escalation/partial/hash.test

```diff
@@ -0,0 +1,10 @@
+table_create Users TABLE_NO_KEY
+column_create Users name COLUMN_SCALAR ShortText
+table_create Terms TABLE_HASH_KEY ShortText --default_tokenizer TokenDelimit
+column_create Terms index COLUMN_INDEX Users name
+load --table Users
+[
+{"name": "Alice"},
+{"name": "Bob"}
+]
+select Users --filter 'name @ "Alice "'
```

Comment:
@komainu8  LGTM except:

* please squash 2 commit  into 1 commit.
* it seems better to include actual error in commit message, too.

```
2019-08-01 14:47:17.367852|i| exact: 0
2019-08-01 14:47:17.367920|i| unsplit: 0
2019-08-01 14:47:17.367937|e| [table][search][hash] invalid mode: suffix
...
2019-08-01 14:47:17.368328|i| partial: 0
```